substrate-node/pallets/pallet-smart-contract/src/storage_state/v9.rs

```diff
@@ -0,0 +1,978 @@
+            Weight::zero()
+        }  
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+        info!("current pallet version: {:?}", PalletVersion::<T>::get());
+        assert!(PalletVersion::<T>::get() == types::StorageVersion::V8 || PalletVersion::<T>::get() == types::StorageVersion::V9);
+
+        check_pallet_smart_contract::<T>();
+
+        Ok(vec![])
+    }
+}
+
+pub fn check_pallet_smart_contract<T: Config>() {
+    info!("💥💥💥💥💥 CHECKING PALLET SMART CONTRACT STORAGE 💥💥💥💥💥");
+    check_contracts::<T>();
+    check_contracts_to_bill_at::<T>();
+    check_active_node_contracts::<T>();
+    check_active_rent_contract_for_node::<T>();
+    check_contract_id_by_node_id_and_hash::<T>();
+    check_contract_id_by_name_registration::<T>();
+    check_contract_lock::<T>();
+    check_solution_providers::<T>();
+    check_contract_billing_information_by_id::<T>();
+    check_node_contract_resources::<T>();
+}
+
+pub fn clean_pallet_smart_contract<T: Config>(current_stage: MigrationStage
+) -> (frame_support::weights::Weight, Option<MigrationStage>) {
+    info!("🧼 Cleaning Smart Contract pallet storage [{}/10]", current_stage);
+    match current_stage {
+        0 => (Weight::zero(), Some(current_stage + 1)),
+        1 => (clean_contracts::<T>(), Some(current_stage + 1)),
+        2 => (clean_contracts_to_bill_at::<T>(), Some(current_stage + 1)),
+        3 => (clean_active_node_contracts::<T>(), Some(current_stage + 1)),
+        4 => (clean_active_rent_contract_for_node::<T>(), Some(current_stage + 1)),
+        5 => (clean_contract_id_by_node_id_and_hash::<T>(), Some(current_stage + 1)),
+        6 => (clean_contract_id_by_name_registration::<T>(), Some(current_stage + 1)),
+        7 => (clean_contract_lock::<T>(), Some(current_stage + 1)),
+        8 => (clean_solution_providers::<T>(), Some(current_stage + 1)),
+        9 => (clean_contract_billing_information_by_id::<T>(), Some(current_stage + 1)),
+        // Last cleaning operation, set stage to none to stop migration
+        10 => {
+            let weight = clean_node_contract_resources::<T>();
+            PalletVersion::<T>::put(types::StorageVersion::V9);
+            info!("🔔 Ending Smart Contract pallet storage cleaning");
+            (weight.add(1), None)
+        }
+        // Should never happen
+        _ => (Weight::zero(), None),
+    }
+}
+
+// Contracts
+pub fn check_contracts<T: Config>() {
+    debug!(
+        "🔎  Smart Contract pallet {:?} checking Contracts storage map START",
+        PalletVersion::<T>::get()
+    );
+
+    let contract_id_range = 1..=ContractID::<T>::get();
+
+    for (contract_id, contract) in Contracts::<T>::iter() {
+        if contract_id != contract.contract_id {
+            debug!(
+                " ⚠️    Contracts[id: {}]: wrong id ({})",
+                contract_id, contract.contract_id
+            );
+        }
+        if !contract_id_range.contains(&contract_id) {
```

Comment:
> I'm not sure I get why you are checking if the id is in the range. It's always the case no?

Indeed, it should always be the case that s why I am checking it to guaranty that this assumption is always true.

> And if it would not be the case why not do: ...

Because I also want to check `contract_id >= 0` so using the range I check both at the same time with same complexity I guess

Reply:
But ContractID is u64 so it can never be < 0... It will always be >= 0

Reply:
Indeed we still want to identify `contract_id == 0` case

Reply:
Yes, I could have done 
~~~rust
if contract_id == 0 || contract_id > ContractID::<T>::get() {
  // log
}
~~~
but I found it more elegant to do 
~~~rust
let contract_id_range = 1..=ContractID::<T>::get();
if !contract_id_range.contains(&contract_id) {
  // log
}
~~~
I think complexity is same, correct?